File: erpnext/support/page/contractor_qr_scan/contractor_qr_scan.js
frappe.pages['contractor-qr-scan'].on_page_load = function(wrapper) {
	var page = frappe.ui.make_app_page({
		parent: wrapper,
		title: 'All Agnikul Contractors can mark their attendance here',
		single_column: true
	});
	$(frappe.render_template("contractor_qr_scan",{
		// qr_code:"Hiiiiii SVR"
	})).appendTo(page.body);

	let $btn = page.set_primary_action('Go home',() => window.location.href ="http://127.0.0.1:8000/app/agkl-contractors")
	// frappe.ready(function () {
	// 	frappe.msgprint("Helloooo")
	// });

		let $Entry_btn = document.getElementById("Entry").addEventListener("click", function ()  {
			videoOverlay.style.display = "block";
			const scanner = new Instascan.Scanner({ video: document.getElementById('preview') });

			scanner.addListener('scan', function (content) {
				// Redirect to the result page with the scanned data as a query parameter
				// window.location.href = '/result?data=' + content;
				frappe.call({
					method: 'erpnext.support.page.contractor_qr_scan.contractor_qr_scan.qr_code_tx',
					args: {
						name_worker: content,
						entry_or_exit: "Entry",


					},
					callback: function (r) {
						frappe.msgprint(r.message)
						setTimeout(function () {
							window.location.href = 'http://127.0.0.1:8000/app/agkl-contractors';

						}, 2000);
					}
				});
			});

			Instascan.Camera.getCameras().then(function (cameras) {
				if (cameras.length > 0) {
					scanner.start(cameras[0]);
				} else {
					console.error('No cameras found.');
				}
			}).catch(function (e) {
				console.error(e);
			});
		})
		let $Exit_btn = document.getElementById("Exit").addEventListener("click",function ()  {
			videoOverlay.style.display = "block";
			
			const scanner = new Instascan.Scanner({ video: document.getElementById('preview') });

			scanner.addListener('scan', function (content) {
				// Redirect to the result page with the scanned data as a query parameter
				// window.location.href = '/result?data=' + content;
				frappe.call({
					method: 'erpnext.support.page.contractor_qr_scan.contractor_qr_scan.qr_code_tx',
					args: {
						name_worker: content,
						entry_or_exit: "Exit",

					},
					callback: function (r) {
						frappe.msgprint(r.message)
						setTimeout(function () {
							window.location.href = 'http://127.0.0.1:8000/app/agkl-contractors';

						}, 2000);

					}
				});
			});

			Instascan.Camera.getCameras().then(function (cameras) {
				if (cameras.length > 0) {
					scanner.start(cameras[0]);
				} else {
					console.error('No cameras found.');
				}
			}).catch(function (e) {
				console.error(e);
			});
		})
		videoOverlay.addEventListener("click", function() {
            videoOverlay.style.display = "none";
        });

		
	// let field = page.add_field({
	// 	label:'attendance',
	// 	fieldtype: 'Button',
	// 	fieldname: 'attendance_cont',
	// action() {
	// 		console.log(1);
	// 	}
	// });
}

File: erpnext/support/page/contractor_qr_scan/contractor_qr_scan.py
import frappe

@frappe.whitelist()
def qr_code_tx(name_worker,entry_or_exit):
    if entry_or_exit=="Entry":
        if(frappe.db.exists("Contractors",name_worker)):
            if (frappe.db.exists("Contractor Attendance",name_worker+'-'+frappe.utils.nowdate())):
                return "Already marked for the day"
            else:
                doc = frappe.new_doc("Contractor Attendance")
                doc.worker_code=name_worker
                if doc.in_time==None:
                    doc.in_time=frappe.utils.now().split()[1][:5]
                
                doc.insert()
        else:
            return "Please check QR"
        
        return "Attendance Marked"
			
    else:
        if(frappe.db.exists("Contractors",name_worker)):
            if (frappe.db.exists("Contractor Attendance",name_worker+'-'+frappe.utils.nowdate())):				
                doc = frappe.get_doc("Contractor Attendance", name_worker+'-'+frappe.utils.nowdate())
                if doc.out_time==None:
                    doc.out_time=frappe.utils.now().split()[1][:5]
                else:
                    return "Already marked for the day"
                doc.save()
                return "Attendance Marked"
            else:
                return "Please mark entry first"
        else:
            return "Please check QR"
    # return qr_code